browserdebuggertools/chrome/interface.py

```diff
@@ -107,17 +107,29 @@ def execute_javascript(self, script):
         return result.get("value")
 
     def get_url(self):
-        return self.execute_javascript("document.URL")
+        # type: () -> str
+        """
+        Consider enabling the Page domain to increase performance.
+
+        :returns: The url of the current page.
+        """
+        return self._socket_handler.event_handlers["PageLoad"].get_current_url()
 
     def get_document_readystate(self):
         """ Gets the document.readyState of the page.
         """
         return self.execute_javascript("document.readyState")
 
     def get_page_source(self):
-        """ Returns a string serialization of the active document's DOM
+        # type: () -> str
         """
-        return self.execute_javascript("document.documentElement.innerHTML")
+        Consider enabling the Page domain to increase performance.
+
+        :returns: A string serialization of the active document's DOM.
+        """
+        root_node_id = self._socket_handler.event_handlers["PageLoad"].get_root_node_id()
+
+        return self.execute("DOM", "getOuterHTML", {"backendNodeId": root_node_id})["outerHTML"]
```

Comment:
In get_url(), the logic is implemented in the PageLoadEventHandler.
Couldn't it also give you the page source?
Further, perhaps the interface could be a bit neater if you hide self._socket_handler.event_handlers["PageLoad"] behind a property